### ISSUE TITLE
[Firefox] popup, background color of items ?

### ISSUE THREAD

**Issue by Betsy25** [uBlock-LLC/uBlock#1093]
Since a had a bad system crash & a system restore, when I now build to the latest commit https://github.com/gorhill/uBlock/commit/9ee9ceed7df95b9c08864bd36718ad5b7f905955 , I suddenly see no more fully background colored items but rather a very small bar at the beginning of the item ?

Is this normal ?
If so, which commit is responsible please ? (And what is the point in making it harder to properly distinguish between blocked/allowed items ?)

Firefox stable + Firefox nightly, Win7

http://i.imgur.com/UKUozoO.png

**Comment by gorhill** [comment]
Yes it's normal, it's how I wanted it to be originally.

**Comment by chrisaljoudi** [comment]
> And what is the point in making it harder to properly distinguish between blocked/allowed items ?

While I understand what you're saying, I personally disagree.

The colors were previously being used in overzealous abundance that made the table visually "noisy", making it more difficult to read and more effortful to visually parse. 

Accent colors are most effective and powerful when used sparingly. 

---

One would not say a rainbow is _strongly red_, because it's equally _strongly every other color, too_ with no base color that's easy to detect.

**Comment by harshanvn** [comment]
i like the newer solution. It is elegant!!

**Comment by chrisaljoudi** [comment]
@Betsy25 

More thoughts. :)

An example (though simplistic) manifestation of this is the somewhat-frequently-talked-about ["60-30-10 rule"](http://www.tigercolor.com/color-lab/tips/tip-01.html). It states that if you a have a three-color palette/scheme, your base color should take up 60% of the "space", secondary 30%, and your accent color 10%.

Now obviously this is really rudimentary and is not in any way the magic single answer. There are many other variables to consider. 

You can see, however, that given we need more than just 3 colors for uBlock's UI all of which need to be distinct, the relative frequency of the usage of the colors needs to be allocated wisely. That kind of consideration reduces "mental load" needed for the user to figure out what's going on every time the UI is presented to them.

**Comment by gorhill** [comment]
I've never been happy with the previous solution, I went with it originally because I had problem to make the current one work at the time: the colored indicators would end up under the scroll bar on Chromium. Eventually I just got tired to try to make it work and just colored the whole cell but I really never liked it. I knew I would come back to it.

When I tried again the bring back the original idea to life again last week, it worked right away, because now the scroll bar is in the parent container, not the rows' container. I have withheld the changes for two weeks to fine tune the size/color/saturation and settled the current recipe.
